webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/RemoveNodeService.scala

```diff
@@ -513,6 +514,18 @@ class RemoveNodeServiceImpl(
 }
 
 
+class RemoveFromCache(nodeInfoService: NodeInfoService) extends PostNodeDeleteAction {
+  override def run(nodeId: NodeId, mode: DeleteMode, info: Option[NodeInfo], status: Set[InventoryStatus]): UIO[Unit] = {
+    if (mode == DeleteMode.Erase) {
+      NodeLoggerPure.Delete.debug(s"  - delete node from NodeInfoService Cache'${nodeId.value}'") *>
```

Comment:
I think we should do that in all cases and remove the logic that looks in the `removed` branch in the cache `isUpToDate` method

Reply:
I think that removing the check in removed branch would fail, as NodeInfoService also looks in this branch for getDeletedNodeInfos (even if not using the cache for that); at least not in this PR